compiler/rustc_macros/src/query.rs

```diff
@@ -432,6 +447,11 @@ pub fn rustc_queries(input: TokenStream) -> TokenStream {
                 $macro!(#cached_queries);
             }
         }
+        macro_rules! rustc_feedable_queries {
+            ( $macro:ident! ) => {
+                $macro!(#feedable_queries);
+            }
+        }
```

Comment:
please avoid generating new macro_rules macros in the proc-macro - you can add a `(feedable)` token to the modifiers instead, which lets the calling code generate its own macro-rules macro if need be. Emitting it in rustc_query isn't extensible because you have less information available when using it than in `rustc_query_append`.

Reply:
This change is from https://github.com/rust-lang/rust/pull/96840